Client/Screen.py

```diff
@@ -171,73 +153,102 @@ def onreturn_play_menu_input_ip(self, val):  # check onreturn of playMenu IP inp
+                self.__screenState = sstatecons.ScreenState.MAIN_MENU
+                self.__connectionMenu.disable()
 
-                self.connectingMenu.disable()
-                self.init_connecting_menu()  # reinitialize connectingMenu so that it flushes its msgs
+                # reinitialize. connectionMenu so that it flushes msgs
+                [self.__connectionMenu, self.__connectionMenuState] = self._init_connection_menu()
 
                 # acknowledged the connection error status, and now set back the flag to None
                 Client.Client.get_instance().connection_alive = None
-                self.is_conn_msg_shown = False
 
                 pg.time.wait(2500)
 
-        elif Client.Client.get_instance().connection_alive is not None:
-            self.connectingMenu.add_line('')
-
-            if Client.Client.get_instance().connection_alive:
-                self.connectingMenu.disable()
-                self.init_connecting_menu()
-                self.connectingMenu.add_line('Successfully connected.')
-            else:
-                self.connectingMenu.add_line('Connection error, please try again!')
-
-            self.is_conn_msg_shown = True
-
-    def connecting_menu_player_count_selector_onchange(self, txt_value, value):
-        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.CHANGE_PLAYER_NUMBER,
-                                          target=climess.Target.GAME)
+    def _connection_menu_player_count_selector_onchange(self, _, value):
+        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.CHANGE_PLAYER_NUMBER, target=climess.Target.GAME)
         msg.new_number = value
         Client.Client.get_instance().send_message(msg)
 
-    def connecting_menu_start_pressed(self):
-        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.START_GAME_MANUALLY,
-                                      target=climess.Target.GAME)
+    def _connection_menu_start_pressed(self):
+        msg = BaseMessage.BaseMessage(mess_type=climess.MessageType.START_GAME_MANUALLY, target=climess.Target.GAME)
         Client.Client.get_instance().send_message(msg)
 
-    def game_screen(self):
-        self.screen.fill((0, 0, 0))  # black bg
+    #  More useful functions below!!!
+    def update(self, events):
+        # draw the adequate screen (according to the state)
+        if self.__screenState == sstatecons.ScreenState.MAIN_MENU:
+            if not self.__mainMenu.is_enabled():
+                self.__mainMenu.enable()
+            self.__mainMenu.mainloop(events)
+        elif self.__screenState == sstatecons.ScreenState.CONNECTION_MENU:
+            if not self.__connectionMenu.is_enabled():
+                self.__connectionMenu.enable()
+            self.__connectionMenu.mainloop(events)
+        elif self.__screenState == sstatecons.ScreenState.GAME:
+            self._game_screen()
```

Comment:
draw adequate screen function?

Reply:
excuse me :D ?

Reply:
Make a function for these lines and just call itfrom update